cpp/oneapi/dal/algo/kmeans_init/detail/compute_ops_dpc.cpp

```diff
@@ -24,20 +24,23 @@ namespace v1 {
 
 using dal::detail::data_parallel_policy;
 
-template <typename Float, typename Method, typename Task>
-struct compute_ops_dispatcher<data_parallel_policy, Float, Method, Task> {
-    compute_result<Task> operator()(const data_parallel_policy& ctx,
+template <typename Policy, typename Float, typename Method, typename Task>
+struct compute_ops_dispatcher<Policy, Float, Method, Task> {
+    compute_result<Task> operator()(const Policy& ctx,
                                     const descriptor_base<Task>& params,
                                     const compute_input<Task>& input) const {
         using kernel_dispatcher_t = dal::backend::kernel_dispatcher<
             KERNEL_SINGLE_NODE_CPU(backend::compute_kernel_cpu<Float, Method, Task>),
-            KERNEL_SINGLE_NODE_GPU(backend::compute_kernel_gpu<Float, Method, Task>)>;
+            KERNEL_UNIVERSAL_SPMD_GPU(backend::compute_kernel_gpu<Float, Method, Task>)>;
         return kernel_dispatcher_t{}(ctx, params, input);
     }
 };
 
-#define INSTANTIATE(F, M, T) \
-    template struct ONEDAL_EXPORT compute_ops_dispatcher<data_parallel_policy, F, M, T>;
+#define INSTANTIATE(F, M, T)                                                \
+    template struct ONEDAL_EXPORT                                           \
+        compute_ops_dispatcher<dal::detail::data_parallel_policy, F, M, T>; \
+    template struct ONEDAL_EXPORT                                           \
+        compute_ops_dispatcher<dal::detail::spmd_data_parallel_policy, F, M, T>;
```

Comment:
What is expected in case of SPMD call for a method w/o SMPD support?

Reply:
Discussed